sheepdog/transactions/upload/sub_entities.py

```diff
@@ -282,9 +284,21 @@ def _register_index(self):
         if self.urls:
             urls.extend(self.urls)
 
+        consent_codes = self.node._props.get("consent_codes")
+        authz = []
+        if consent_codes:
+            authz = ["/consents/" + code for code in consent_codes]
+        else:
```

Comment:
so it's either consent code or `/programs/{}/projects/{}` - is there no use case for having both?

Reply:
if there's a consent code in `authz`, the `acl` field will still contain program-project: how does that work?

Reply:
- There's not a case for having both at the moment, the list in authz combines conjunctively (you have access only if you have access to all the entries), so it should be one or the other
- ACL should work exactly the same way as it did for backwards-compatibility—it's not really used in the new auth system